src/OrchardCore.Modules/OrchardCore.Users/Controllers/EmailConfirmationController.cs

```diff
@@ -52,6 +61,9 @@ public async Task<IActionResult> ConfirmEmail(string userId, string code)
 
         if (result.Succeeded)
         {
+            var userContext = new UserContext(user);
+            await _userEventHandlers.InvokeAsync((handler, context) => handler.ConfirmedAsync(userContext), userContext, _logger);
```

Comment:
```suggestion
            await _userEventHandlers.InvokeAsync((handler, context) => handler.ConfirmedAsync(userContext), UserContext(user), _logger);
```

Reply:
I ended up changing this back. There was a compiler error because `userContext` is additionally passed into `handler.ConfirmAsync(...)`. The other places where `_userEventHandlers` is used look to do a [similar thing](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Users/Drivers/UserDisplayDriver.cs#L125).